connect/src/com/telenor/connect/headerenrichment/HeFlowDecider.java

```diff
@@ -77,13 +77,22 @@ public static Uri chooseFlow(Uri uri, Context context) {
             }
             return uri;
         } else {
-            boolean isMobile = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)
-                    .isConnectedOrConnecting();
-            boolean isWifi = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
-                    .isConnectedOrConnecting();
+            boolean isMobile = false;
+            if (connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE) != null) {
```

Comment:
How about something like this to save number of lines and make the code easier to read:

boolean isMobile = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE) != null
                   && connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE).isConnectedOrConnecting();

Reply:
@ulfcarlin refactored